### PR TITLE
Replace some GPDB code with functionally equivalent upstream code.

### PR DESCRIPTION
**Pull request by hlinnaka**
There's no particular urgency for this, but less code to maintain is good.

### CODE REVIEW

File: src/backend/cdb/motion/ic_tcp.c [hunk 1]
```diff
@@ -2245,18 +2245,20 @@ void
...
 	for (i = 1; i < nfds; i++)
 	{
 		if (MPP_FD_ISSET(i, fds))
-			appendStringInfo(buf, "%d,", i);
+		{
+			if (!first)
+				appendStringInfoChar(buf, ',');
+			appendStringInfo(buf, "%d", i);
+			first = false;
+		}
```
Comment (BaiShaoqi) [inline src/backend/cdb/motion/ic_tcp.c]: ideal code smell！

File: src/backend/executor/nodeMaterial.c [hunk 2]
```diff
@@ -345,8 +345,8 @@ ExecInitMaterial(Material *node, EState *estate, int eflags)
 	 * index into range table, while the material may refer to the same relation as "outer" varno)
 	 * [JIRA: MPP-25365]
 	 */
-	insist_log(list_length(node->plan.targetlist) == list_length(outerPlan->targetlist),
-			"Material operator does not support projection");
+	if (list_length(node->plan.targetlist) != list_length(outerPlan->targetlist))
+		elog(ERROR, "Material operator does not support projection");
```
Comment (danielgustafsson) [inline src/backend/executor/nodeMaterial.c]: Wan't to lowercase "material" here while at it?

**Comment by hlinnaka** [comment]
On 17/12/2018 15:40, Daniel Gustafsson wrote:
> danielgustafsson approved this pull request.
> 
> LGTM
> 
>> @@ -345,8 +345,8 @@ ExecInitMaterial(Material *node, EState *estate, int eflags)
>   	 * index into range table, while the material may refer to the same relation as "outer" varno)
>   	 * [JIRA: MPP-25365]
>   	 */
> -	insist_log(list_length(node->plan.targetlist) == list_length(outerPlan->targetlist),
> -			"Material operator does not support projection");
> +	if (list_length(node->plan.targetlist) != list_length(outerPlan->targetlist))
> +		elog(ERROR, "Material operator does not support projection");
> 
> Wan't to lowercase "material" here while at it?

I think that's correct in upper-case, actually. I read Material here as 
the name of the operator, and we tend to write those in upper case. 
Compare with "Aggref found in non-Agg plan node" and some other errors 
we have.

- Heikki

**Comment by danielgustafsson** [comment]
> On 17 Dec 2018, at 14:48, Heikki Linnakangas <notifications@github.com> wrote:

> > Wan't to lowercase "material" here while at it?
> 
> I think that's correct in upper-case, actually. I read Material here as 
> the name of the operator, and we tend to write those in upper case. 
> Compare with "Aggref found in non-Agg plan node" and some other errors 
> we have.

Yeah, looking at it again I agree, you’re right.

**Comment by hlinnaka** [comment]
Pushed, thanks for the review!

**Pull request merged**